include/rocksdb/compressor.h

```diff
@@ -0,0 +1,302 @@
+                  uint64_t _sample_for_compression)
+      : opts_(_opts),
+        context_(_context),
+        dict_(_dict),
+        type_(_type),
+        sample_for_compression_(_sample_for_compression) {}
+
+  const CompressionOptions& options() const { return opts_; }
+  const CompressionContext& context() const { return context_; }
+  const CompressionDict& dict() const { return dict_; }
+  CompressionType type() const { return type_; }
+  uint64_t SampleForCompression() const { return sample_for_compression_; }
+};
+
+class UncompressionContext {
+ private:
+  // Compressor-specific context, if the compressor supports the functionality,
+  // nullptr otherwise. A Compressor is responsible for allocating the
+  // underlying memory in the CreateUncompressionContext method and deallocating
+  // it in the DestroyUncompressionContext method.
+  void* context_ = nullptr;
+
+  // Compressor for the selected compression type
+  std::shared_ptr<Compressor> compressor_ = nullptr;
+
+ public:
+  explicit UncompressionContext(CompressionType type);
+  ~UncompressionContext();
+  UncompressionContext(const UncompressionContext&) = delete;
+  UncompressionContext& operator=(const UncompressionContext&) = delete;
+  void* GetContext() const { return context_; };
+};
+
+class UncompressionInfo {
+  const UncompressionContext& context_;
+  const UncompressionDict& dict_;
+  const CompressionType type_;
+
+ public:
+  UncompressionInfo(const UncompressionContext& _context,
+                    const UncompressionDict& _dict, CompressionType _type)
+      : context_(_context), dict_(_dict), type_(_type) {}
+
+  const UncompressionContext& context() const { return context_; }
+  const UncompressionDict& dict() const { return dict_; }
+  CompressionType type() const { return type_; }
+};
+
+// Interface for each compression algorithm to implement.
+class Compressor {
+ public:
+  virtual ~Compressor() = default;
+
+  // Unique name for the compressor.
+  // The name is used to specify the compressor in the options string and for
+  // querying CompressorRegistry.
+  virtual const char* Name() const = 0;
+
+  // Whether the compressor is supported.
+  // For example, a compressor can implement this method to verify its
+  // dependencies or environment settings.
+  virtual bool Supported() const { return true; }
+
+  // Whether the compressor supports dictionary compression.
+  virtual bool DictCompressionSupported() const { return false; }
+
+  // Compress data.
+  // @param info Pointer to CompressionInfo object (containing context,
+  // dictionary, options).
+  // @param compress_format_version Compressed block format.
+  // @param input Buffer containing data to compress.
+  // @param input_length Length of the input data.
+  // @param output Compressed data.
+  // Returns OK if compression completed correctly.
+  // Returns other status in case of error (e.g., Corruption).
+  virtual Status Compress(const CompressionInfo* info,
```

Comment:
This is addressed in the latest code. CompressionInfo doesn't include a Compressor or CompressionContext. In particular, if a compressor needs to carry context (such as ZSTDCompressor), it is responsible for managing it internally rather than through a generic CompressionContext class.